packages/slate/src/interfaces/node.ts

```diff
@@ -222,7 +241,7 @@ export const Node = {
         }
       }
 
-      delete r.selection
+      if (Editor.isEditor(r)) delete r.selection
```

Comment:
whats this for? is this fixing some other bug?

Reply:
Same reason as below: Including the type-guard means TypeScript won't complain anymore since the `selection` key is not guaranteed on any `Node` other than an `Editor`.